lib/rules/max-nesting-depth/__tests__/index.mjs

```diff
@@ -359,6 +359,73 @@ testRule({
 	],
 });
 
+testRule({
+	ruleName,
+	config: [1, { ignoreRules: ['&'], ignore: ['pseudo-classes'] }],
+
+	accept: [
+		{
+			code: stripIndent`
+			a {
+				color: red;
+				& {
+					color: blue;
+				}
+				b {
+					color: gray;
+					& {
+						color: white;
+					}
+					&:hover {
+						color: orange;
+					}
+					&,
+					&:hover {
+						color: purple;
+					}
+				}
+			}
+		`,
```

Comment:
[suggestion] Minor. To save lines and increase visibility, how about making this change? (although from my preference)

```suggestion
			code: stripIndent`
				a {
					color: red;
					& { color: blue; }
					b {
						color: gray;
						& { color: white; }
						&:hover { color: orange; }
						&, &:hover { color: purple; }
					}
				}
			`,
```